src/tool_formparse.c

```diff
@@ -73,7 +72,7 @@ static struct tool_mime *tool_mime_new_data(struct tool_mime *parent,
   if(data) {
     m = tool_mime_new(parent, TOOLMIME_DATA);
     if(!m)
-      CONST_FREE(data);
+      free((char *)data);
```

Comment:
Instead of freeing pointers to const I would rather we change the function prototypes:

~~~diff
diff --git a/src/tool_formparse.c b/src/tool_formparse.c
index 14189d4..c54f9ef 100644
--- a/src/tool_formparse.c
+++ b/src/tool_formparse.c
@@ -64,7 +64,7 @@ static struct tool_mime *tool_mime_new_parts(struct tool_mime *parent)
 }
 
 static struct tool_mime *tool_mime_new_data(struct tool_mime *parent,
-                                            const char *data)
+                                            char *data)
 {
   struct tool_mime *m = NULL;
 
@@ -72,7 +72,7 @@ static struct tool_mime *tool_mime_new_data(struct tool_mime *parent,
   if(data) {
     m = tool_mime_new(parent, TOOLMIME_DATA);
     if(!m)
-      free((char *)data);
+      free(data);
     else
       m->data = data;
   }
@@ -80,7 +80,7 @@ static struct tool_mime *tool_mime_new_data(struct tool_mime *parent,
 }
 
 static struct tool_mime *tool_mime_new_filedata(struct tool_mime *parent,
-                                                const char *filename,
+                                                char *filename,
                                                 bool isremotefile,
                                                 CURLcode *errcode)
 {
@@ -94,7 +94,7 @@ static struct tool_mime *tool_mime_new_filedata(struct tool_mime *parent,
     if(filename) {
       m = tool_mime_new(parent, TOOLMIME_FILE);
       if(!m)
-        free((char *)filename);
+        free(filename);
       else {
         m->data = filename;
         if(!isremotefile)
~~~

/cc @monnerat

Reply:
> Instead of freeing pointers to const I would rather we change the function prototypes:

Agreed